src/main/java/net/fabricmc/accesswidener/AccessWidenerReader.java

```diff
@@ -18,23 +18,52 @@
 
 import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Locale;
+import java.util.regex.Pattern;
 
 public final class AccessWidenerReader {
-	private final Visitor visitor;
+	public static final Charset ENCODING = StandardCharsets.UTF_8;
+
+	// Also includes some weirdness such as vertical tabs
+	private static final Pattern V1_DELIMITER = Pattern.compile("\\s+");
+	// Only spaces or tabs
+	private static final Pattern V2_DELIMITER = Pattern.compile("[ \\t]+");
+	// Prefix used on access types to denote the entry should be inherited by mods depending on this mod
+	private static final String TRANSITIVE_PREFIX = "transitive-";
+
+	// Access widener format versions
+	private static final int V1 = 1;
+	private static final int V2 = 1;
+
+	private final AccessWidenerVisitor visitor;
 
 	private int lineNumber;
 
-	public AccessWidenerReader(AccessWidener accessWidener) {
-		this.visitor = accessWidener;
+	public AccessWidenerReader(AccessWidenerVisitor visitor) {
+		this.visitor = visitor;
 	}
 
-	public AccessWidenerReader(Visitor visitor) {
-		this.visitor = visitor;
+	public void read(byte[] content) {
+		read(content, null);
+	}
+
+	public void read(byte[] content, String currentNamespace) {
+		String strContent = new String(content, ENCODING);
+
+		try {
+			read(new BufferedReader(new StringReader(strContent)), currentNamespace);
+		} catch (IOException e) {
```

Comment:
the method should throw the IOException

Reply:
But why? The caller isn't passing a stream or reader that would throw. Any IOException being thrown here is a bug in the reader, not the caller's fault. (StringReader never throws, BufferedReader should only throw on programming bugs).

Reply:
Ok, sure 👍 its not going to happen anyway.